rv_histogram: Make it more verbose

ChinYikMing opened the pull request:
Section headers must be present for static analysis since the tool uses the `SHT_PROGBITS` type and `SHF_EXECINSTR` flag, which are defined in section headers, to determine the executable machine instructions of the target program

Closed #272

jserv commented on tools/rv_histogram.c:
```diff
@@ -231,21 +231,32 @@ static void insn_hist_incr(const rv_insn_t *ir)
     if (!elf_open(e, elf_prog)) {
-        fprintf(stderr, "elf_open failed\n");
-        return 1;
+        (void) fprintf(stderr, "elf_open %s failed\n", elf_prog);
```
Replace the message with something like "Failed to open"

jserv reviewed (changes requested):
Explain why ELF section headers are necessary for static analysis in the git commit message.

jserv commented on tools/rv_histogram.c:
```diff
@@ -317,5 +328,6 @@ int main(int argc, const char *args[])
 
-    return 0;
+end:
+    return !!ret_val;
```
Simply return `ret_val`.

jserv commented on tools/rv_histogram.c:
```diff
@@ -231,21 +231,32 @@ static void insn_hist_incr(const rv_insn_t *ir)
         print_usage(args[0]);
-        return 1;
+        ret_val = 1;
+        goto end;
```
Instead of using `goto`, you can use `return` instead.

jserv replied:
Alternatively, define a function named `fatal`, which dumps the message and invokes [exit](https://man7.org/linux/man-pages/man3/exit.3.html).

ChinYikMing replied:
MISRA checker reported that single function should have only one return statement. So I made this change. Shall I stick with early return?

jserv replied:
You can safely ignore the warnings reported by Cppcheck at the moment. I shall suppress some of them.

ChinYikMing committed:
rv_histogram: Make it more verbose

Section headers must be present for static analysis since the tool uses
the SHT_PROGBITS type and SHF_EXECINSTR flag, which are defined in
section headers, to determine the executable machine instructions
of the target program

Closed #272

jserv commented:
Thank @ChinYikMing for contributing!
